packages/api-nest/src/repositories/reading-progress.repository.ts

```
@@ -0,0 +1,109 @@
+  }
+
+  /**
+   * Upsert reading progress using INSERT ... ON CONFLICT UPDATE
+   * More efficient than separate find + save operations
+   */
+  async upsertProgress(
+    userId: string,
+    libraryItemId: string,
+    contentVersion: string | null,
+    lastSeenSentinel: number,
+    highestSeenSentinel: number,
```

ESLint

[error] Delete `,`

(prettier/prettier)